EOF on empty files with a language filter

Just tested it on a small repo:

```sh
$ mkdir foo; cd foo; git init
$ touch foo.py; git add foo.py; git commit -m 1
$ echo foo > foo.py; git add foo.py; git commit -m 2
$ hercules.linux_amd64 --burndown --burndown-people --languages-python .
 0 / 5 [-----------------------------------------------------------------------]
[ERROR] 2019/04/09 12:36:21 TreeDiff failed on commit #1 (2) 4b5b10d421d173787db9cb757881889f05d77e84: EOF
2019/04/09 12:36:21 failed to run the pipeline: EOF
```

Looks like it doesn't like empty files (or commits with those).

Re: the "integrity error" with the private repo I've mentioned above - in that commit it was actually kind of the same story -- a Python file was modified which had previously been empty (but existed in the repo), the integrity error crash was at the commit where it was modified.

_Originally posted by @aldanor in https://github.com/src-d/hercules/issues/270#issuecomment-481215737_

--- a/internal/plumbing/tree_diff.go
+++ b/internal/plumbing/tree_diff.go
@@ -278,7 +278,7 @@ func (treediff *TreeDiff) checkLanguage(name string, blobHash plumbing.Hash) (bo
 	}
 	buffer := make([]byte, 1024)
 	n, err := reader.Read(buffer)
-	if err != nil {
+	if err != nil && (blob.Size != 0 || err != io.EOF) {
 		return false, err
 	}
 	if n < len(buffer) {

--- a/internal/plumbing/tree_diff_test.go
+++ b/internal/plumbing/tree_diff_test.go
@@ -263,8 +263,13 @@ func TestTreeDiffFork(t *testing.T) {
 
 func TestTreeDiffCheckLanguage(t *testing.T) {
 	td := fixtureTreeDiff()
-	td.Languages["Go"] = true
 	lang, err := td.checkLanguage(
+		"version.go", plumbing.NewHash("975f35a1412b8ae79b5ba2558f71f41e707fd5a9"))
+	assert.NoError(t, err)
+	assert.True(t, lang)
+	td.Languages["go"] = true
+	delete(td.Languages, allLanguages)
+	lang, err = td.checkLanguage(
 		"version.go", plumbing.NewHash("975f35a1412b8ae79b5ba2558f71f41e707fd5a9"))
 	assert.NoError(t, err)
 	assert.True(t, lang)
@@ -282,10 +287,20 @@ func TestTreeDiffConsumeEnryFilter(t *testing.T) {
 
 	newDiffs := td.filterDiffs(diffs)
 	assert.Len(t, newDiffs, 2)
-	td.Configure(map[string]interface{}{
+	assert.NoError(t, td.Configure(map[string]interface{}{
 		ConfigTreeDiffEnableBlacklist:     true,
 		ConfigTreeDiffBlacklistedPrefixes: []string{"whatever"},
-	})
+	}))
 	newDiffs = td.filterDiffs(diffs)
 	assert.Len(t, newDiffs, 0)
+}
+
+func TestTreeDiffCheckLanguageEmpty(t *testing.T) {
+	td := fixtureTreeDiff()
+	td.Languages["python"] = true
+	delete(td.Languages, allLanguages)
+	lang, err := td.checkLanguage(
+		"__init__.py", plumbing.NewHash("e69de29bb2d1d6434b8b29ae775ad8c2e48c5391"))
+	assert.NoError(t, err)
+	assert.True(t, lang)
 }

